src/nvim/ex_cmds2.c

```diff
@@ -3181,7 +3186,7 @@ int do_source(char_u *fname, int check_other, int is_vimrc)
     }
     si = &SCRIPT_ITEM(current_SID);
     si->sn_name = fname_exp;
-    fname_exp = NULL;
+    fname_exp = vim_strsave(si->sn_name);  // used for autocmd
```

Comment:
Why is this enough to pass the build?

Reply:
`si` is part of `script_items`. When `ga_grow(&script_items,...)` reallocates (see stacktrace above), `si->sn_name` pointer is invalid. Creating a copy here avoids that.